woocommerce/woocommerce-android
# Prevent all cards to be disabled

Part of: #10702

### Why
As part of Expanded Analytics Hub milestone 2, we plan to add support for customizing the cards we display on the Analytic Hub screen.

### Description
This PR supports disabling the last selected card to ensure that at least one card should be selected in the Customize Analytics screen. It does that by introducing the `checkVisibleCards() `  function and creating the new `AnalyticCardConfigurationUI` class to handle analytics cards' enable/disable status.

### Testing instructions
TC1
1. Open the app
2. Tap on See more to navigate to the Analytic Hub screen
3. Tap on the pencil icon to navigate to the Customize Analytics screen
4. Deselect all cards
5. Check that the last card can't be deselected and its status is disabled
6. Select one of the unselected cards
7. Check that the previous disabled card turns its status to enabled and can be unselected
8. Save the configuration with only 1 card selected
9. Tap on the pencil to enter the  Customize Analytics screen again
10. Check that the screen opens with the configuration from step 8 and that there is only one card selected and disabled.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/44989287-048a-44cf-862c-6c2d6a172ccf
